app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+                            const filtered = filterJoinedChannels(joinedChannels, data);
+                            if (filtered?.length) {
+                                setSharedChannels([...sharedChannels, ...filtered]);
+                            }
+                            if (data?.length && !filtered?.length) {
+                                doGetChannels();
+                            } else {
+                                setLoading(false);
+                            }
+                            break;
+                        }
+                        case ARCHIVED:
+                        default:
+                            archivedPage.current += 1;
+                            nextArchived.current = Boolean(data?.length);
+                            if (data?.length) {
+                                setArchivedChannels([...archivedChannels, ...data]);
+                            }
+                            setLoading(false);
+
+                            break;
+                    }
+                }
+            };
+            switch (typeOfChannels) {
+                case PUBLIC:
+                    if (nextPublic.current) {
+                        setLoading(true);
+                        fetchChannels(
+                            serverUrl,
+                            currentTeamId,
+                            publicPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || [])); // Handle error?
+                    }
+                    break;
+                case SHARED:
+                    if (nextShared.current) {
+                        setLoading(true);
+                        fetchSharedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            sharedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+                case ARCHIVED:
+                default:
+                    if (canShowArchivedChannels && nextArchived.current) {
+                        setLoading(true);
+                        fetchArchivedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            archivedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+            }
+        }
+    }, [loading, typeOfChannels, joinedChannels, archivedChannels, channels, sharedChannels, serverUrl, currentTeamId]);
+
+    const selectActiveChannels = useCallback(() => {
```

Comment:
why does this need to be memoized?

Reply:
selectActiveChannels is a dependency of searchChannels, which is a callback passed to the SearchBar component. Now that I look at it again, I would say it makes more sense to use it as a useMemo, instead of a function, but still should be memoized.

Reply:
What happens if you don't memoize it and have it just as a regular function?

Reply:
Now that it is a value, and not a function, we can "not memoize" it, since the possible values are state values. Will do.